Changes several tests so that they still pass with new libmesh feature

DanielYankura opened the pull request:
<!--
If this PR is associated with an issue be sure to reference it
by including "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).

If this PR implements an enhancement that does not have an existing issue, please add the following:
-->

## Reason
<!--Why do you need this feature or what is the enhancement?-->
- Currently if a sideset and nodeset have the same ID libmesh will merge the two when making a nodeset from that sideset
- New libmesh feature throws and error instead
- Several MOOSE tests will fail when this gets fixed

## Design
<!--A concise description (design) of the enhancement.-->
- In named_entities.e and named_entities.xda nodesets were renumbered to avoid this
- The AdvancedExtruderGenerator boundary swap test is commented out until the associated libmesh PR is merged 

## Impact
<!--Will the enhancement change existing APIs or add something new?-->
- MOOSE tests won't fail with new libmesh PR

<!--
If this PR implements a bug fix, please create an issue for the bug and reference the issue.
-->
Must be merged before libMesh/libmesh#4121
refs #30085

DanielYankura committed:
Changes several tests so that they still pass with new libmesh feature

- New libmesh feature throws and error when making a nodeset from a sideset if a nodeset with same ID exists
- named_entities meshes were changes to avoid this
- BoundaryInfo function now called in AdvancedExtruderGenerator that tells which sidesets should be excluded from this check

closes #30085

GiudGiud commented on framework/src/meshgenerators/AdvancedExtruderGenerator.C:
```diff
@@ -491,10 +491,13 @@ AdvancedExtruderGenerator::generate()
                                    _boundary_swap_pairs[e].count(id_to_copy)
                                        ? _boundary_swap_pairs[e][id_to_copy]
                                        : id_to_copy);
+            boundary_info.add_equivalent_sideset(_boundary_swap_pairs[e][id_to_copy]);
```
ok the PR says why but let s have a comment this will be needed elsewhere

DanielYankura committed:
Changes so that tests won't fail before libmesh PR is merged

- removed call to add_equivalent_sideset from AdvancedExtruderGenerator.C
- added skip to corresponding extruder boudary swap test
- Two named_entities mesh files left the same

DanielYankura commented:
So @roystgnr recommended that I try to make my upgrade path a bit smoother. So I updated this PR so that it just updates the boundary IDs for two meshes and comments out one test that would otherwise fail once the associated libmesh PR is merged. That way after that PR is merge no tests should be failing and I can open another PR for MOOSE that uncomments out that one test and adds the function call needed to AdvancedExtruderGenerator.C.

GiudGiud commented on test/tests/meshgenerators/advanced_extruder_generator/tests:
use skip = ' number of the issue' instead

GiudGiud commented:
Let’s not commit the back and forth to the repo

DanielYankura commented:
Just squashed my last two commits!

GiudGiud commented on test/tests/meshgenerators/advanced_extruder_generator/tests:
```diff
@@ -120,6 +120,7 @@
     requirement = 'The system shall be able to retain and swap boundary ids during extrusion.'
     design = 'AdvancedExtruderGenerator.md'
     issues = '#21852'
+    skip = 'Awaiting PR to fix libmesh PR'
```
```suggestion
    skip = 'Requires libMesh PR #12345'
```

DanielYankura commented:
Just squashed that in as well!

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2811201/), step Docs: sync website on 4e20226 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/30291/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2811216/), step Generate coverage on 4e20226 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>f54877</th>
      <th colspan=4>#30291 4e2022</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.30%</td>
      <td>85.29%</td>
      <td>-0.01%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>109952</td>
      <td>109941</td>
      <td>-11</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>18952</td>
      <td>18963</td>
      <td>+11</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/30291/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/30291/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/30291/coverage/xfem)

</details>



This comment will be updated on new commits.
